File: 08_Day/DicMap.js
function processData(input) {
    //Enter your code here
    const data = input.split('\n');
    const n = Number(data.shift());
    const phoneBookData = new Map();
    const objData = data.reduce((target, item, index) => {
        index < n ? phoneBookData.set(item.split(' ')[0], item.split(' ')[1]) : target.push(item);

        return target;
    }, []);

    objData.forEach((val, index) => { 
        console.log(phoneBookData.get(val) ? `${val}=${phoneBookData.get(val)}` : 'Not found');
    })
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: 08_Day/DicMap.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
# Enter your code here. Read input from STDIN. Print output to STDOUT
import sys
data = [line.rstrip() for line in sys.stdin.readlines()]
n = int(data[0])
dic = {}

for d in data[1:n+1]:
    k = d.split(' ')[0]
    v = d.split(' ')[1]
    dic[k] = v

for t in data[n+1:]:
    if t in dic.keys():
        print(t + '=' + dic[t])
    else:
        print('Not found')
